Formula/klee.rb

```diff
@@ -15,24 +15,49 @@ class Klee < Formula
 
   depends_on "cmake" => :build
   depends_on "gperftools"
-  depends_on "llvm@11"
+  depends_on "llvm"
   depends_on "python-tabulate"
   depends_on "python@3.9"
   depends_on "sqlite"
+  depends_on "stp"
   depends_on "wllvm"
   depends_on "z3"
 
   uses_from_macos "zlib"
 
   # klee needs a version of libc++ compiled with wllvm
   resource "libcxx" do
-    url "https://github.com/llvm/llvm-project/releases/download/llvmorg-11.1.0/llvm-project-11.1.0.src.tar.xz"
-    sha256 "74d2529159fd118c3eac6f90107b5611bccc6f647fdea104024183e8d5e25831"
+    url "https://github.com/llvm/llvm-project/releases/download/llvmorg-12.0.0/llvm-project-12.0.0.src.tar.xz"
+    sha256 "9ed1688943a4402d7c904cc4515798cdb20080066efa010fe7e1f2551b423628"
   end
 
-  def install
-    llvm = Formula["llvm@11"]
+  # Patches for LLVM 12 Support
+  # https://github.com/klee/klee/pull/1389
+  patch do
+    url "https://github.com/klee/klee/commit/74ea9e5e63c5933ca2d5d7f846858c4de6e86b81.patch?full_index=1"
+    sha256 "5af19fb3dbc609a180014f89a78bd007316e1384f3b23bf64fcd15621951b130"
+  end
+
+  patch do
+    url "https://github.com/klee/klee/commit/a34fb8961649bf3a065ec8f0eb4bc58715fd1d57.patch?full_index=1"
+    sha256 "beb18d3e74c8a580e2c3785e7224cacfb878b527fc4f261f7acb2ebecec93fb0"
+  end
+
+  patch do
+    url "https://github.com/klee/klee/commit/2b29d86a39421ac76421b888b96613173bc18851.patch?full_index=1"
+    sha256 "34515f7841dc3bc6e68888aa98492e3e003131fdc43018f4923799b0e2ff32fd"
+  end
+
+  patch do
+    url "https://github.com/klee/klee/commit/c0b10c6f7a00d81cfce24115168dd06888685f87.patch?full_index=1"
+    sha256 "d970235981e6f96f408b5943f80877b633a01cf098e1b4be2c19967b5412eff5"
+  end
+
+  def llvm
+    deps.map(&:to_formula).find { |f| f.name.match? "^llvm" }
+  end
```

Comment:
I see this popping up in more formulae, maybe it can be a DSL like `Formulae.any_dep("^llvm")`?

Reply:
Yup, but I'm not sure how to make it useful for the other formulae though, since the others use this slightly differently. Maybe `any_dep` and `all_deps` methods that can take a regex/string argument. I'll try to think about it a bit more.